CheatSheets/java-cheatsheet.md

```diff
@@ -0,0 +1,145 @@
+# Table of Content
+
+ - [Data Types](#data-types)
+ - [Data Conversion](#data-conversion)
+ - [Operators](#operators)
+ - [Statements](#statements)
+ - [String Methods](#string-methods)
+ 
+## Data Types
+| Data Type | Size |
+|--|--|
+| boolean | 1 bit |
+| char | 2 byte |
+| int | 4 byte |
+| short | 2 byte| 
+|long |8 byte |
+|float|4 byte|
+|double|8 byte |
+
+## Data Conversion
+
+### String to Number
+```java
+    int i = Intege­r.p­ars­eInt(_str_);  
+	double d = Double.pa­rse­Double(_str_);
+```
+	
+### Any Type to String
+```java
+	String s = String.va­lueOf(_value_);  
+
+```
+  
+### Numeric Conver­sions
+```java
+	int i = (int) _numeric expression_; 
+
+```
+
+## Operators
+| Operator Category | Operators |
+|--|--|
+| Arithmetic operators |+, -, /, *, %  |
+|Relational operators|<, >, <=, >=,==, !=|
+|Logical operators|&&, \|\||
+|Assignment operator|=, +=, −=, ×=, ÷=, %=, &=, ^=, \|=, <<=, >>=, >>>=|
+|Increment and Decrement operator|++ , - -|
+|Conditional operators|?, :|
+|Bitwise operators|^, &, \||
+|Special operators|. (dot operator to access methods of class)|
+
+## Statements
+### If Statement
+```java
+if ( _expression_ ) {  
+­ _statements_  
+} else if ( _expression_ ) {  
+­ _statements_  
+} else {  
+­ _statements_  
+}  
+```
+### While Loop 
+```java
+while ( _expression_ ) {  
+­ _statements_  
+}  
+```
+
+  
+### Do-While Loop 
+```java
+do {  
+­ _statements_  
+} while ( _expression_ ); 
+```
+  
+### For Loop 
+```java
+for ( int i = 0; i < _max_; ++i) {  
+­ _statements_  
+}  
+```
+
+  
+### For Each Loop
+```java
+for ( _var_ : _collection_ ) {  
+­ _statements_  
+} 
+```
+ 
+  
+### Switch Statement 
+```java
+switch ( _expression_ ) {  
+­ case _value_:  
+­ ­ ­ _statements_  
+­ ­ ­ ­break;  
+­ case _value2_:  
+­ ­ ­ _statements_  
+­ ­ ­ ­break;  
+­ ­def­ault:  
+­ ­ ­ _statements_  
+} 
+```
+  
+### Exception Handling
+```java
+try {  
+­ ­sta­tem­ents;  
+} catch (_Except­ionType_  _e1_) {  
+­ ­sta­tem­ents;  
+} catch (Exception _e2_) {  
+­ ­cat­ch-all statem­ents;  
+} finally {  
+­ ­sta­tem­ents;  
+}
+```
```

Comment:
Is it possible to give them in a `tabular format`? If it is `possible`, then `modify` your code, please.